pallets/gear-program/src/migrations/allocations.rs

```diff
@@ -118,7 +118,7 @@ impl<T: Config> OnRuntimeUpgrade for MigrateAllocations<T> {
                 "incorrect count of elements old {} != new {}",
             );
             ensure!(
-                Pallet::<T>::current_storage_version() == MIGRATE_TO_VERSION,
+                Pallet::<T>::on_chain_storage_version() == MIGRATE_TO_VERSION,
```

Comment:
It shouldn't. Once the `on_chain_version` has been bumped, the migration inside `on_runtime_upgrade` simply won't run anymore.
On the other hand, comparing the `current_storage_version` to the `MIGRATE_TO_VERSION` would fail on the very first run as long as there is a newer migration for the pallet in the same Runtime. Furthermore, it doesn't indicate if a migration has been successful (because it doesn't depend on the storage at all).